python/ray/serialization.py

```diff
@@ -222,7 +222,7 @@ def _deserialize_pickle5_data(self, data):
     def _deserialize_msgpack_data(self, data, metadata):
         msgpack_data, pickle5_data = split_buffer(data)
 
-        if metadata == ray_constants.OBJECT_METADATA_TYPE_PYTHON:
+        if metadata[0:1] == ray_constants.OBJECT_METADATA_TYPE_PYTHON:
```

Comment:
We should just make the metadata a comma-separated list or some other standard format.

Reply:
Fixed!